src/include/common/hip/rpp_hip_math.hpp

```diff
@@ -218,6 +232,19 @@ __device__ __forceinline__ void rpp_hip_math_bitwiseAnd8(d_uchar8 *src1_uc8, d_u
         dst_uc8->uc1[7] = src1_uc8->uc1[7] & src2_uc8->uc1[7];
 }
 
+// Used to do bitwise and of the scaled float image representations - Values scaled from 0 to 255 with constant mask
+__device__ __forceinline__ void rpp_hip_math_scaled_bitwiseAnd8(d_float8 *src_f8, d_uchar8 *src_mask_u8, d_float8 *dst_f8)
+{
+        dst_f8->f1[0] = (float)((uchar)nearbyintf(src_f8->f1[0]) & src_mask_u8->uc1[0]);
```

Comment:
converting float to int will change the precision. Can you try reinterpret_cast like 
__int_as_float((__float_as_int(src_f8->f1[0]) & src_mask_u8->uc1[0]))